src/auth/mas.py

```diff
@@ -146,9 +154,36 @@ def register_user(
         logger.exception("Error registering new user.")
         raise MASRegisterUserFailedError("Error registering new user.") from exc
 
+    provision_all_users(container=container)
     return password
 
 
+def provision_all_users(
+    container: ops.model.Container,
+) -> None:
+    """Start jobs to provision all users.
+
+    Args:
+        container: Synapse container.
+
+    Raises:
+        MASProvisionUserFailedError: when triggering the provisioning jobs fails.
+    """
+    command = [
+        MAS_EXECUTABLE_PATH,
+        "-c",
+        MAS_CONFIGURATION_PATH,
+        "manage",
+        "provision-all-users",
```

Comment:
Does this job have any impact on MAS?
Also, should this be run after every restart too?

Reply:
I'm also not sure if this command is necessary, will confirm with MAS team and let you know.

Reply:
I got an update from MAS team and basically it ensures that MAS users are synced with the synapse DB, although we won't need to run it every time a new user is added as MAS will take care of that. So I removed the command wrapper for now, and potentially add it back as a charm action if necessary.